includes/cli/class-integrity-check.php

```diff
@@ -0,0 +1,629 @@
+		$all_discrepancies = [];
+		$node_columns = [ 'email', 'network_id', 'hub_status' ];
+		
+		foreach ( $discrepancies as $node ) {
+			WP_CLI::line( sprintf( 'Analyzing discrepancies for node: %s', $node->get_url() ) );
+			
+			$node_url = $node->get_url();
+			$node_name = str_replace( [ 'https://www.', 'https://', 'http://www.', 'http://' ], '', $node_url );
+			$node_columns[] = $node_name;
+
+			$specific_discrepancies = self::find_discrepancies_chunked( $hub_data, $node, $chunk_size, $verbose, $max_records );
+
+			// Process discrepancies for this node.
+			foreach ( $specific_discrepancies as $discrepancy ) {
+				$key = $discrepancy['email'] . '::' . $discrepancy['network_id'];
+				
+				if ( ! isset( $all_discrepancies[ $key ] ) ) {
+					$all_discrepancies[ $key ] = [
+						'email'      => $discrepancy['email'],
+						'network_id' => $discrepancy['network_id'],
+						'hub_status' => $discrepancy['hub_status'],
+					];
+				}
+				
+				$all_discrepancies[ $key ][ $node_name ] = $discrepancy['node_status'];
+			}
+		}
+
+		// Display consolidated table if there are any discrepancies.
+		if ( ! empty( $all_discrepancies ) ) {
+			WP_CLI::line( '' );
+			WP_CLI::line( sprintf( 'Found %d total discrepancies:', count( $all_discrepancies ) ) );
+			WP_CLI::line( '' );
+
+			// Prepare table data with node columns.
+			$table_data = [];
+			foreach ( $all_discrepancies as $discrepancy ) {
+				// Fill in missing node statuses with empty string.
+				foreach ( $node_columns as $column ) {
+					if ( ! isset( $discrepancy[ $column ] ) && ! in_array( $column, [ 'email', 'network_id', 'hub_status' ] ) ) {
+						$discrepancy[ $column ] = '';
+					}
+				}
+				$table_data[] = $discrepancy;
+			}
+
+			// Display as table using WP-CLI's table formatter.
+			WP_CLI\Utils\format_items( 'table', $table_data, $node_columns );
+		}
+
+		if ( $fix_discrepancies ) {
+			WP_CLI::line( 'Fix discrepancies functionality would be implemented here.' );
+		}
+	}
+
+
+	/**
+	 * Get membership data from a node via REST API
+	 *
+	 * @param \Newspack_Network\Node\Node $node The node to query.
+	 * @return array Array of (email, status) pairs
+	 */
+	private static function get_node_membership_data( $node ) {
+		$endpoint = sprintf( '%s/wp-json/newspack-network/v1/integrity-check/memberships', $node->get_url() );
+		$endpoint = add_query_arg( [ '_t' => time() ], $endpoint ); // Cache-busting parameter.
+		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get
+		$response = wp_remote_get(
+			$endpoint,
+			[
+				'headers' => $node->get_authorization_headers( 'integrity-check' ),
+				'timeout' => 60, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout
+			]
+		);
+
+		if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			WP_CLI::error( sprintf( 'Failed to get membership data from node: %s', $node->get_url() ) );
```

Comment:
Just noting that `WP_CLI::error` will halt execution. Not sure if this is what we want, or if the intention is to just report the error and continue with other nodes if available.

Reply:
Good question! I'd leave it as it as, opting for "full integrity check". If one node can't give us data, then the whole check is not complete.